apps/judicial-system/backend/src/app/modules/police/police.service.ts

```diff
@@ -171,43 +209,31 @@ export class PoliceService {
       .then(async (res: Response) => {
         if (res.ok) {
           if (this.config.policeCaseApiV2Available) {
-            const response = await res.json()
-
-            return (
-              response.skjol?.map(
-                (file: {
-                  rvMalSkjolMals_ID: string
-                  heitiSkjals: string
-                  malsnumer: string
-                  dagsStofnad: string
-                }) => ({
-                  id: file.rvMalSkjolMals_ID,
-                  name: file.heitiSkjals.endsWith('.pdf')
-                    ? file.heitiSkjals
-                    : `${file.heitiSkjals}.pdf`,
-                  policeCaseNumber: file.malsnumer,
-                  displayDate: file.dagsStofnad,
-                }),
-              ) ?? []
-            )
+            const response: z.infer<typeof this.responseStructure> =
+              await res.json()
+            this.responseStructure.parse(response)
+
+            return response.skjol?.map((file) => ({
+              id: file.rvMalSkjolMals_ID.toString(),
+              name: file.heitiSkjals.endsWith('.pdf')
+                ? file.heitiSkjals
+                : `${file.heitiSkjals}.pdf`,
+              policeCaseNumber: file.malsnumer,
+              displayDate: file.dagsStofnad,
+            }))
           } else {
-            const response = await res.json()
-
-            return response.map(
-              (file: {
-                rvMalSkjolMals_ID: string
-                heitiSkjals: string
-                malsnumer: string
-                dagsStofnad: string
-              }) => ({
-                id: file.rvMalSkjolMals_ID,
-                name: file.heitiSkjals.endsWith('.pdf')
-                  ? file.heitiSkjals
-                  : `${file.heitiSkjals}.pdf`,
-                policeCaseNumber: file.malsnumer,
-                displayDate: file.dagsStofnad,
-              }),
-            )
+            const response: z.infer<typeof this.policeCaseFileStructure>[] =
```

Comment:
I think it would make more sense to vaildate the V2 structure and not the old one. Actually, this else case is being deleted in another PR.